zerver/models.py

```diff
@@ -284,18 +284,30 @@ def name_changes_disabled(realm):
         return settings.NAME_CHANGES_DISABLED
     return settings.NAME_CHANGES_DISABLED or realm.name_changes_disabled
 
+class DuplicateRealmAliasException(Exception):
+    pass
+
 class RealmAlias(models.Model):
     realm = models.ForeignKey(Realm, null=True) # type: Optional[Realm]
     # should always be stored lowercase
     domain = models.CharField(max_length=80, db_index=True) # type: Text
+    subdomains_allowed = models.BooleanField(default=False)
 
-def can_add_alias(domain):
-    # type: (Text) -> bool
-    if settings.REALMS_HAVE_SUBDOMAINS:
+    class Meta(object):
+        unique_together = ("realm", "domain", "subdomains_allowed")
+
+    def save(self, *args, **kwargs):
+        # type: (List[Any], Dict[Any, Any]) -> None
+        if not can_add_alias(self.domain, self.subdomains_allowed):
+            raise DuplicateRealmAliasException("A Realm for %s already exists." % (self.domain))
+        super(RealmAlias, self).save(*args, **kwargs)
+
+def can_add_alias(domain, subdomains_allowed=False):
```

Comment:
With the new policy it would not be possible for an organization to claim both example.com and *.example.com simultaneously, while the previous policy allowed it. Would it be fine?